rasterio/_warp.pyx

```diff
@@ -964,13 +966,15 @@ cdef class WarpedVRTReaderBase(DatasetReaderBase):
 
         cdef GDALRasterBandH hBand = NULL
         src_alpha_band = 0
+        dst_alpha_band = 0
         for bidx in src_dataset.indexes:
             hBand = GDALGetRasterBand(hds, bidx)
             if GDALGetRasterColorInterpretation(hBand) == GCI_AlphaBand:
                 src_alpha_band = bidx
+                dst_alpha_band = bidx
```

Comment:
If think it's safe to assume that if there is a source alpha band, there will be a destination alpha band with the same index.